compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -971,10 +971,12 @@ TR::Register *OMR::X86::TreeEvaluator::baddEvaluator(TR::Node *node, TR::CodeGen
    // target register generated and we return NULL to the caller (which should be
    // a store) to indicate that the store has already been done.
    //
-
+   
+   // Need special handling when Eflags results are needed. Otherwise, the later part of baddEvaluator could optimize
+   // the instruction to a 4 bytes LEA.
    if (NEED_CC(node))
       {
-      TR_ASSERT(node->getOpCodeValue() == TR::badd,
+      TR_ASSERT(node->getOpCodeValue() == TR::badd || node->getOpCodeValue() == TR::buadd,
```

Comment:
buadd is deprecated and should not be handled really - I think it can go from this assert.